lib/account.js

```diff
@@ -43,141 +49,171 @@ function parseMessage(message, callback) {
+        newPassword: request.payload.password
+      }), 'utf8');
+    });
   });
 
-  server.method('account.createUser', function(request, callback) {
-    var createRequest = hyperquest({
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-        'x-ratelimit-ip': getIPAddress(request)
-      },
-      uri: loginAPI + '/api/v2/user/create'
-    });
+  server.method('account.createUser', function(request) {
+    return new Promise(function(resolve, reject) {
+      var createRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/create'
+      });
 
-    createRequest.on('error', callback);
+      createRequest.on('error', reject);
 
-    createRequest.on('response', function(message) {
-      parseMessage(message, function(err, json) {
-        callback(err, json);
+      createRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(json);
+          }
+        });
       });
-    });
 
-    createRequest.end(JSON.stringify({
-      user: {
-        email: request.payload.email,
-        username: request.payload.username,
-        mailingList: request.payload.feedback,
-        prefLocale: request.payload.lang
-      },
-      oauth: {
-        client_id: request.payload.client_id
-      },
-      password: request.payload.password,
-      audience: options.uri
-    }), 'utf8');
-  });
-
-  server.method('account.getUser', function(userId, callback) {
-    var ttl = null;
-    var getRequest = hyperquest({
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      uri: loginAPI + '/user/id/' + userId
+      createRequest.end(JSON.stringify({
+        user: {
+          email: request.payload.email,
+          username: request.payload.username,
+          mailingList: request.payload.feedback,
+          prefLocale: request.payload.lang
+        },
+        oauth: {
+          client_id: request.payload.client_id
+        },
+        password: request.payload.password,
+        audience: options.uri
+      }), 'utf8');
     });
+  });
 
-    getRequest.on('error', function(err) {
-      callback(err, null, 0);
-    });
+  server.method('account.getUser', function(userId, flags) {
+    return new Promise(function(resolve, reject) {
+      var getRequest = hyperquest({
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        uri: loginAPI + '/user/id/' + userId
+      });
 
-    getRequest.on('response', function(message) {
-      parseMessage(message, function(err, json) {
-        if (json && !json.user) {
-          ttl = 0;
-        }
+      getRequest.on('error', function(err) {
+        flags.ttl = 0;
```

Comment:
Since server methods no longer support callbacks, you can't pass the ttl for the server method via the callback. Instead, according to Hapi docs, each server method receives a `flags` property that I believe you can set the ttl (and some other stuff) on. I'm pretty sure that's how it works, but no idea how to test it.